crates/starknet-server/src/api/http/endpoints/postman.rs

```diff
@@ -1,30 +1,158 @@
 
 pub(crate) async fn postman_load(
-    Json(_l1_contract): Json<PostmanLoadL1MessagingContract>,
-) -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+    Extension(state): Extension<HttpApiHandler>,
+    Json(data): Json<PostmanLoadL1MessagingContract>,
+) -> HttpApiResult<Json<MessagingLoadAddress>> {
+    let mut starknet = state.api.starknet.write().await;
+
+    let messaging_contract_address = starknet
+        .configure_messaging(
+            &data.network_url,
+            data.address.as_deref(),
+            data.private_key.as_deref(),
+        )
+        .await
+        .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?;
+
+    Ok(Json(MessagingLoadAddress { messaging_contract_address }))
 }
 
-pub(crate) async fn postman_flush() -> HttpApiResult<()> {
-    Err(HttpApiError::GeneralError)
+pub(crate) async fn postman_flush(
+    Extension(state): Extension<HttpApiHandler>,
+    Json(data): Json<FlushParameters>,
+) -> HttpApiResult<Json<FlushedMessages>> {
+    // Need to handle L1 to L2 first in case that those messages
+    // will create L2 to L1 messages.
+    let mut starknet = state.api.starknet.write().await;
+
+    let dry_run = data.dry_run.unwrap_or(false);
+
+    let messages_to_l2 = if dry_run {
+        Ok(vec![])
+    } else {
+        starknet
+            .fetch_and_execute_messages_to_l2()
+            .await
+            .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?
+            .into_iter()
+            .map(|m| m.try_into())
+            .collect::<Result<Vec<MessageToL2>, HttpApiError>>()
+    };
+
+    let from_block = if let Some(m) = &starknet.messaging { m.last_local_block } else { 0 };
+
+    let (messages_to_l1, last_local_block) = if dry_run {
+        (
+            starknet
+                .collect_messages_to_l1(from_block)
+                .await
+                .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?
+                .into_iter()
+                .map(|m| m.try_into())
+                .collect::<Result<Vec<MessageToL1>, HttpApiError>>(),
+            0,
+        )
+    } else {
+        let (msgs, b) = starknet
+            .collect_and_send_messages_to_l1(from_block)
+            .await
+            .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?;
+
+        (
+            msgs.into_iter()
+                .map(|m| m.try_into())
+                .collect::<Result<Vec<MessageToL1>, HttpApiError>>(),
+            b,
+        )
+    };
+
+    let l1_provider = if dry_run {
+        "dry run".to_string()
+    } else {
+        starknet.messaging_url().unwrap_or("Not set".to_string())
+    };
+
+    if !dry_run {
+        // +1 to ensure this last block is not collected anymore.
+        starknet.messaging.as_mut().expect("Messaging expected configured").last_local_block =
+            last_local_block + 1;
+    }
+
+    match (messages_to_l1, messages_to_l2) {
+        (Ok(l1s), Ok(l2s)) => {
```

Comment:
Why does this need to be four cases? Can't we first process `messages_to_l1`, then `messages_to_l2`?

Reply:
I removed this it was not necessary as you mentioned.